lib_comfyui/comfyui_reverse_proxy.py

```diff
@@ -0,0 +1,58 @@
+import multiprocessing
+import sys
+import shutil
+import subprocess
+import threading
+import time
+import socket
+import importlib
+from lib_comfyui import argv_conversion, webui_settings
+importlib.reload(argv_conversion)
+importlib.reload(webui_settings)
+
+
+process = None
+
+
+def start():
+    global process
+
+    comfyui_argv = list(sys.argv)
+    argv_conversion.set_comfyui_argv(comfyui_argv)
+    port = comfyui_argv[comfyui_argv.index('--port') + 1]
+
+    local_tunnel = shutil.which('lt')
+    if local_tunnel is None:
+        return
+
+    def on_create_local_tunnel(int_port):
+        global process
+        if process is not None:
+            return
+
+        while True:
+            time.sleep(0.5)
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            result = sock.connect_ex(('127.0.0.1', int_port))
+            if result == 0:
+                break
+            sock.close()
+
+        print("Launching localtunnel...")
+        process = subprocess.Popen([local_tunnel, "--port", port, "--local-host", '127.0.0.1'], stdout=subprocess.PIPE)
+        for line in process.stdout:
+            line = line.decode()
+            if line.startswith('your url is:'):
+                webui_settings.set_comfyui_url(line.split('your url is:')[1].strip())
+            print(line, end='')
+
+    threading.Thread(target=on_create_local_tunnel, args=(int(port),), daemon=True).start()
+
+
+def stop():
+    global process
```

Comment:
really basic question: can this process and the other process in comfyui_adapter interfere with each other because they are both globals?

Reply:
huh, python globals aren't shared across modules. What the heck, python xD
"Each file has its own global variable"
 https://towardsdatascience.com/common-mistakes-when-dealing-with-multiple-python-files-b4f4dc4d5643#:~:text=Each%20file%20has%20its%20own%20global%20variable

But yeah, that makes sense now. All good.